src/lib/operators/projection.cpp

```diff
@@ -211,8 +214,10 @@ std::shared_ptr<const Table> Projection::_on_execute() {
   for (auto expression_id = ColumnID{0}; expression_id < expression_count; ++expression_id) {
     const auto& expression = expressions[expression_id];
     if (const auto pqp_column_expression = std::dynamic_pointer_cast<PQPColumnExpression>(expression)) {
-      const auto& original_id = pqp_column_expression->column_id;
-      input_column_to_output_column[original_id] = expression_id;
+      if (forwardable_pqp_columns.contains(expression)) {
```

Comment:
Why do we still need to cast in the line before?

Reply:
Well, we need the cast to get the `column_id`.

Reply:
Ah, right.